benchexec/tools/vampire.py

```diff
@@ -0,0 +1,128 @@
+                if run.output.text.startswith("Parsing Error"):
+                    return "PARSING ERROR"
+            if run.exit_code.value == 1:
+                # Error during portfolio mode
+                if run.output.text.startswith("% Exception at proof search level\n"):
+                    if any(line.startswith("Parsing Error") for line in run.output):
+                        return "PARSING ERROR"
+                # Not really an error but unable to finish
+                reasons = self.get_other_termination_reasons(run.output)
+                if reasons == ["Time limit"]:
+                    return "TIMEOUT"
+                if reasons == ["Memory limit"]:
+                    return "OUT OF MEMORY"
+                if reasons == ["Refutation not found, incomplete strategy"]:
+                    return "INCOMPLETE"
+                if reasons == ["Refutation not found, non-redundant clauses discarded"]:
+                    return "INCOMPLETE"
+            # Some other error
+            return result.RESULT_ERROR
+        else:
+            # Successfully finished
+            if status in self.SZS_UNSAT:
+                return result.RESULT_TRUE_PROP
+            elif status in self.SZS_SAT:
+                return result.RESULT_FALSE_PROP
+            else:
+                return result.RESULT_UNKNOWN
+
+    # SZS status values that Vampire returns
+    SZS_UNSAT = ["ContradictoryAxioms", "Theorem", "Unsatisfiable"]
+    SZS_SAT = ["CounterSatisfiable", "Satisfiable"]
+    SZS_FAIL = ["Timeout", "GaveUp", "User"]
+
+    def get_szs_status(self, output):
+        """
+        Extract the SZS status from the output.
+        @param output: The output of the tool as instance of class RunOutput.
+        @return a non-empty string, or None if no unique SZS status was found
+        """
+        status = None
+        for line in output:
+            if line.startswith("% SZS status"):
+                if status is None:
+                    words = line.split()
+                    status = words[3] if len(words) >= 4 else None
+                else:
+                    # More than one SZS status => this is an error!
+                    return None
+        return status
+
+    def get_other_termination_reasons(self, output):
+        """
+        Extract termination reasons from the output from lines starting with '% Termination reason'.
+        @param output: The output of the tool as instance of class RunOutput.
+        @return List of strings, the termination found in the output (without prefix).
+        """
+        prefix = "% Termination reason: "
+        reasons = []
+        for line in output:
+            if line.startswith(prefix):
+                reasons.append(line[len(prefix) :])
+        return reasons
+
+    def get_value_from_output(self, output, identifier):
+        if identifier == "szs-status":
+            return self.get_szs_status(output) or "--"
```

Comment:
Note that if you just return `None` in case the identifier is not found, BenchExec will handle this gracefully (e.g., empty value in CSV output, long dash rendered in HTML tables).

Reply:
One more thing:
An additional `return None` is required by our style guide at the end of this function, it should not just fall through to the end because there is also a return value in some branch.